src/Avalonia.Base/Platform/Storage/FileIO/BclLauncher.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Avalonia.Compatibility;
+using Avalonia.Metadata;
+
+namespace Avalonia.Platform.Storage.FileIO;
+
+internal class BclLauncher : ILauncher
+{
+    public virtual Task<bool> LaunchUriAsync(Uri uri)
```

Comment:
A random thought here: The Uri scheme can be used to indicate `file`. https://learn.microsoft.com/en-us/dotnet/api/system.uri.scheme?view=net-8.0. Perhaps LaunchUriAsync should be a little smarter and detect file scheme? Or is relying on the OS through Process.Start considered enough in most cases.

Reply:
Potentially, yes.
.NET frameworks (UWP, MAUI...) tend to separate these two methods. As well as mobile APIs always use two different approaches for Uris and Files (Android, iOS, Tizen do). 
So, I though it might be a safe future proofing to do the same. Just in case. 

At the same time, Qt doesn't separate them. Qt has a single method, where you can pass "file://" link.